--- a/arch/arm/boot/dts/suniv-f1c100s.dtsi
+++ b/arch/arm/boot/dts/suniv-f1c100s.dtsi
@@ -141,6 +141,15 @@
 				pins = "PA2", "PA3";
 				function = "uart1";
 			};
+      spi0_pc_pins: spi0-pc-pins {
+        pins = "PC0", "PC1", "PC2", "PC3";
+        function = "spi0";
+      };
+
+      spi1_pa_pins: spi1-pa-pins {
+        pins = "PA0", "PA1", "PA2", "PA3";
+        function = "spi1";
+      };
 			lcd_rgb666_pins: lcd-rgb666-pins {
 				pins = "PD0", "PD1", "PD2", "PD3", "PD4",
 				       "PD5", "PD6", "PD7", "PD8", "PD9",
@@ -223,6 +232,31 @@
 			resets = <&ccu RST_BUS_UART2>;
 			status = "disabled";
 		};
+    spi0: spi@1c05000 {
+      compatible = "allwinner,suniv-f1c100s-spi",
+             "allwinner,sun8i-h3-spi";
+      reg = <0x01c05000 0x1000>;
+      interrupts = <10>;
+      clocks = <&ccu CLK_BUS_SPI0>, <&ccu CLK_BUS_SPI0>;
+      clock-names = "ahb", "mod";
+      resets = <&ccu RST_BUS_SPI0>;
+      status = "disabled";
+      #address-cells = <1>;
+      #size-cells = <0>;
+    };
+
+    spi1: spi@1c06000 {
+      compatible = "allwinner,suniv-f1c100s-spi",
+             "allwinner,sun8i-h3-spi";
+      reg = <0x01c06000 0x1000>;
+      interrupts = <11>;
+      clocks = <&ccu CLK_BUS_SPI1>, <&ccu CLK_BUS_SPI1>;
+      clock-names = "ahb", "mod";
+      resets = <&ccu RST_BUS_SPI1>;
+      status = "disabled";
+      #address-cells = <1>;
+      #size-cells = <0>;
+    };
 		usb_otg: usb@1c13000 {
 			compatible = "allwinner,suniv-f1c100s-musb";
 			reg = <0x01c13000 0x0400>;
--- a/arch/arm/boot/dts/suniv-f1c100s-licheepi-nano.dts
+++ b/arch/arm/boot/dts/suniv-f1c100s-licheepi-nano.dts
@@ -13,12 +13,12 @@
 
 
 	aliases {
-		serial1 = &uart1;
+		serial0 = &uart0;
 	};
 
 
 	chosen {
-		stdout-path = "serial1:115200n8";
+		stdout-path = "serial0:115200n8";
 	};
 
 	panel: panel {
@@ -78,11 +78,51 @@
 &otg_sram {
 	status = "okay";
 };
-&uart1 {
+&uart0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_pa_pins>;
+	pinctrl-0 = <&uart0_pe_pins>;
 	status = "okay";
 };
+&spi0 {
+  pinctrl-names = "default";
+  pinctrl-0 = <&spi0_pc_pins>;
+  status = "okay";
+  flash@0 {
+    #address-cells = <1>;
+    #size-cells = <1>;
+    compatible = "winbond,w25q128", "jedec,spi-nor";
+    reg = <0>;
+    spi-max-frequency = <50000000>;
+    partitions {
+      compatible = "fixed-partitions";
+      #address-cells = <1>;
+      #size-cells = <1>;
+
+      partition@0 {
+        label = "u-boot";
+        reg = <0x000000 0x100000>;
+        read-only;
+      };
+
+      partition@100000 {
+        label = "dtb";
+        reg = <0x100000 0x10000>;
+        read-only;
+      };
+
+      partition@110000 {
+        label = "kernel";
+        reg = <0x110000 0x4f0000>;
+        read-only;
+      };
+
+      partition@600000 {
+        label = "rootfs";
+        reg = <0x600000 0xA00000>;
+      };
+    };
+  };
+};
 &usb_otg {
 	dr_mode = "peripheral";	// peripheral | otg | host
 	status = "okay";
